roles/destroy_vms/tasks/main.yml

```diff
@@ -1,32 +1,44 @@
-- name: Remove network
+- name: Create inventory host to vm name mapper
+  set_fact:
+    cluster_vm_names: "{{ (cluster_vm_names | default([])) + [item.startswith(cluster_name) | ternary(item, (cluster_name + '_' + item))] }}"
+  loop: "{{ groups['masters'] + groups['workers'] | default([]) }}"
+  when: hostvars[item]['vendor'] | lower == 'kvm'
+
+- debug: # noqa unnamed-task
+    var: vm_to_inventory_mapper
+    verbosity: 1
+
+- name: Destroy vms
+  import_tasks: destroy_networks.yml
   become: true
-  ignore_errors: yes
+
+- name: Get VMs to Remove
   block:
-    - name: destroy network
-      community.libvirt.virt_net:
-        name: "{{ network_name }}"
-        command: destroy
-
-    - name: undefine network
-      community.libvirt.virt_net:
-        name: "{{ network_name }}"
-        command: undefine
-
-- name: Delete existing bridges (if any)
-  community.general.nmcli:
-    conn_name: "{{ item }}"
-    type: bridge
-    state: absent
-  loop:
-    - "{{ bridge_name }}"
-    - "{{ vm_bridge_interface }}_{{ bridge_name }}"
-  ignore_errors: yes
-  become: true
-  when: (SETUP_VM_BRIDGE | default(true)) | bool == true
+    - name: List vms
+      community.libvirt.virt:
+        command: list_vms
+      register: vms_list
+      become: true
 
-- name: Destroy pool
-  import_tasks: destroy_pools.yml
-  ignore_errors: yes
+    - name: VM list
+      debug:
+        var: vms_list.list_vms
+        verbosity: 1
+
+    - name: Get filtered vm list
+      set_fact:
+        vms_to_remove: "{{ (vms_to_remove | default([])) + [item] }}"
+      loop: "{{ vms_list.list_vms }}"
+      when: item in cluster_vm_names
+
+- name: VMS to remove
+  debug:
+    var: vms_to_remove
+    verbosity: 1
+
+- name: Clean VM images
+  import_tasks: clean_vm_images.yml
```

Comment:
Is `command: undefine` not sufficient? https://libvirt.org/manpages/virsh.html#undefine suggests that an undefine (with the options all turned on) is complete enough, and https://github.com/ansible-collections/community.libvirt/blob/7cc0b5ff881fc2be108df7158e68a99057c8c952/plugins/modules/virt.py#L385 implies that the `command: undefine` option will remove everything.

Reply:
Poorly explained: if `command: undefine` does not, as its docstring implies, "wipe it from the face of the earth", then a `virsh undefine --remove-all-storage --nvram` would be preferable for achieving the same result with less code than custom plays for extracting and removing volumes.

Reply:
You we're correct thanks for the spot :)